packages/nuxt/src/pages/utils.ts

```diff
@@ -94,6 +101,12 @@ export function generateRoutesFromFiles (files: string[], pagesDir: string): Nux
   return prepareRoutes(routes)
 }
 
+const PAGE_META_NAME_RE = /definePageMeta\(\s*{[^}]*name:\s*(["'`])((?:[^"'`\\]|\\.)+?)\1/
```

Comment:
I think this is too permissive.

For example, this matches:

```ts
definePageMeta({
 something: () => ({ name: 'test' })
})
```

I think we probably would need to parse the file.